### PR TITLE
raftkv: implement reverse_seek

### PR DESCRIPTION
**Pull request by BusyJay**
@ngaut @siddontang @disksing PTAL

### CODE REVIEW

File: src/raftstore/coprocessor/region_snapshot.rs [hunk 1]
```diff
@@ -42,22 +46,47 @@ impl<'a> RegionSnapshot<'a> {
     }
 
     // TODO: return TakeWhile instead.
-    fn new_iterator(&self, start_key: &[u8]) -> DBIterator {
+    fn new_iterator<'b>(&'b self, start_key: &[u8]) -> Box<Iterator<Item = Kv> + 'b> {
```
Comment (siddontang) [inline src/raftstore/coprocessor/region_snapshot.rs]: why not using common `'a` ?

**Comment by siddontang** [comment]
LGTM

**Comment by BusyJay** [comment]
@ngaut PTAL

File: src/raftstore/coprocessor/region_snapshot.rs [hunk 2]
```diff
@@ -42,22 +46,47 @@ impl<'a> RegionSnapshot<'a> {
     }
 
     // TODO: return TakeWhile instead.
```
Comment (ngaut) [inline src/raftstore/coprocessor/region_snapshot.rs]: staled comments ?

**Comment by ngaut** [comment]
LGTM

**Pull request merged**